tests/cancun/eip7516_blobgasfee/test_blobgasfee_opcode.py

```diff
@@ -46,13 +46,20 @@ def callee_address(pre: Alloc, callee_code: Bytecode) -> Address:
     return pre.deploy_contract(callee_code)
 
 
+@pytest.fixture
+def store_slot() -> Bytecode:
+    """Storage slot where to save the expected value, by default 1."""
+    return Op.PUSH1(1)
+
+
 @pytest.fixture
 def caller_code(
     call_gas: int,
     callee_address: Address,
+    store_slot: Bytecode,
 ) -> Bytecode:
     """Bytecode used to call the bytecode containing the BLOBBASEFEE opcode."""
-    return Op.SSTORE(Op.NUMBER, Op.CALL(gas=call_gas, address=callee_address))
+    return Op.SSTORE(store_slot, Op.CALL(gas=call_gas, address=callee_address))
```

Comment:
One trick could be to use `Op.SELFBALANCE` instead of `Op.NUMBER`, and then add `value=1` (1 wei) to the transaction, so that:
- Each transaction increases the balance of the contract by 1.
- When the contract uses its own balance as the key, each transaction it will save to the next key incrementally because of the 1 wei included with the transaction.
- This is compatible with fill and execute because we no longer rely on the block number, and the contract balance always starts at zero on both instances.

Reply:
Thanks for the suggestion. I've applied it and it works for both `fill` and `execute`.

However, AFAIU CI is still failing in the same spot as https://github.com/ethereum/execution-spec-tests/pull/2266#issuecomment-3370418432.